salt/modules/zypperpkg.py

```diff
@@ -1387,11 +1397,20 @@ def refresh_db(force=None, root=None):
     salt.utils.pkg.clear_rtag(__opts__)
     ret = {}
     refresh_opts = ["refresh"]
+    global_opts = []
     if force is None:
         force = __pillar__.get("zypper", {}).get("refreshdb_force", True)
     if force:
         refresh_opts.append("--force")
-    out = __zypper__(root=root).refreshable.call(*refresh_opts)
+    repos = kwargs.get("repos", [])
+    refresh_opts.extend([repos] if not isinstance(repos, list) else repos)
+
+    if kwargs.get("gpgautoimport", False):
+        global_opts.append("--gpg-auto-import-keys")
+
+    out = __zypper__(root=root).refreshable.call(
+        *global_opts, *refresh_opts, success_retcodes=[0, 6]
+    )
```

Comment:
I think it would be nice to add here a comment to clarify why those retcodes:
```suggestion
   # We do the actual call to zypper refresh.
   # We ignore retcode 6 which is returned when there are no repositories defined.
    out = __zypper__(root=root).refreshable.call(
        *global_opts, *refresh_opts, success_retcodes=[0, 6]
    )
```